lib/tapioca/generator.rb

```diff
@@ -624,5 +631,37 @@ def purge_stale_dsl_rbi_files(files)
         say("")
       end
     end
+
+    sig { void }
+    def perform_sync_verification
+      diff = {}
+
+      removed_rbis.each do |gem_name|
+        filename = existing_rbi(gem_name)
+        diff[filename] = :removed
+      end
+
+      added_rbis.each do |gem_name|
+        filename = expected_rbi(gem_name)
+        diff[filename] = gem_rbi_exists?(gem_name) ? :changed : :added
+      end
+
+      if diff.empty?
+        say("Nothing to do, all RBIs are up-to-date.")
+      else
+        say("RBI files are out-of-date. In your development environment, please run:", :green)
+        say("  `#{Config::DEFAULT_COMMAND} sync`", [:green, :bold])
+        say("Once it is complete, be sure to commit and push any changes", :green)
+
+        say("")
+
+        say("Reason:", [:red])
+        diff.group_by(&:last).sort.each do |cause, diff_for_cause|
+          say(build_error_for_files(cause, diff_for_cause.map(&:first)))
+        end
+
+        exit(1)
```

Comment:
This is nearly identical to a block of code in the DSL, only difference is the suggested command (`sync` here, `dsl` in the DSL one). I considered generalizing to a separate method but decided against it to keep the diff smaller.

Reply:
Very similar to [`verify_dsl_rbi`](https://github.com/Shopify/tapioca/blob/b8e7ceeaead2df91b847051e4c8d8b35dbf842d4/lib/tapioca/generator.rb#L545) indeed. Can we extract it as a helper so it can be shared instead of duplicating it?